control_data_collecting_tool/scripts/data_collecting_base_node.py

```diff
@@ -0,0 +1,159 @@
+        )
+
+        self.declare_parameter(
+            "wheel_base",
+            2.79,
+            ParameterDescriptor(description="Wheel base [m]"),
+        )
+
+        self.sub_odometry_ = self.create_subscription(
+            Odometry,
+            "/localization/kinematic_state",
+            self.onOdometry,
+            1,
+        )
+
+        self.sub_acceleration_ = self.create_subscription(
+            AccelWithCovarianceStamped,
+            "/localization/acceleration",
+            self.onAcceleration,
+            1,
+        )
+
+        self.operation_mode_subscriber_ = self.create_subscription(
+            OperationModeState,
+            "/system/operation_mode/state",
+            self.subscribe_operation_mode,
+            10,
+        )
+
+        self._present_kinematic_state = None
+        self._present_acceleration = None
+        self.present_operation_mode_ = None
+
+        """
+        velocity and acceleration grid
+        velocity and steer grid
+        """
+        self.num_bins_v = self.get_parameter("NUM_BINS_V").get_parameter_value().integer_value
+        self.num_bins_steer = (
+            self.get_parameter("NUM_BINS_STEER").get_parameter_value().integer_value
+        )
+        self.num_bins_a = self.get_parameter("NUM_BINS_A").get_parameter_value().integer_value
+        self.v_min, self.v_max = (
+            self.get_parameter("V_MIN").get_parameter_value().double_value,
+            self.get_parameter("V_MAX").get_parameter_value().double_value,
+        )
+        self.steer_min, self.steer_max = (
+            self.get_parameter("STEER_MIN").get_parameter_value().double_value,
+            self.get_parameter("STEER_MAX").get_parameter_value().double_value,
+        )
+        self.a_min, self.a_max = (
+            self.get_parameter("A_MIN").get_parameter_value().double_value,
+            self.get_parameter("A_MAX").get_parameter_value().double_value,
+        )
+
+        self.collected_data_counts_of_vel_acc = np.zeros(
+            (self.num_bins_v, self.num_bins_a), dtype=np.int32
+        )
+        self.collected_data_counts_of_vel_steer = np.zeros(
+            (self.num_bins_v, self.num_bins_steer), dtype=np.int32
+        )
+
+        self.v_bins = np.linspace(self.v_min, self.v_max, self.num_bins_v + 1)
+        self.steer_bins = np.linspace(self.steer_min, self.steer_max, self.num_bins_steer + 1)
+        self.a_bins = np.linspace(self.a_min, self.a_max, self.num_bins_a + 1)
+
+        self.v_bin_centers = (self.v_bins[:-1] + self.v_bins[1:]) / 2
+        self.steer_bin_centers = (self.steer_bins[:-1] + self.steer_bins[1:]) / 2
+        self.a_bin_centers = (self.a_bins[:-1] + self.a_bins[1:]) / 2
+
+    def onOdometry(self, msg):
+        self._present_kinematic_state = msg
+
+    def onAcceleration(self, msg):
+        self._present_acceleration = msg
+
+    def subscribe_operation_mode(self, msg):
+        self.present_peration_mode_ = msg.mode
```

Comment:
```suggestion
        self.present_operation_mode_ = msg.mode
```

Reply:
could you please some typos?

Reply:
Thank you for your feedback. I have made the changes in [c95cdfb](https://github.com/autowarefoundation/autoware_tools/pull/134/commits/c95cdfb26f5ff18858b7fb1a74dc769845b28b98).

Reply:
I rebased on main so please force push if you fix

Reply:
Thank you for repeatedly pointing out my mistakes. I have made the corrections.